Guides/Keyed.md

```diff
@@ -0,0 +1,78 @@
+# Indexed
+
+[[Source](https://github.com/apple/swift-algorithms/blob/main/Sources/Algorithms/Keyed.swift) | 
+ [Tests](https://github.com/apple/swift-algorithms/blob/main/Tests/SwiftAlgorithmsTests/KeyedTests.swift)]
+
+Stores the elements of a sequence as the values of a Dictionary, keyed by the result of the given closure.
+
+```swift
+let fruits = try! ["Apple", "Banana", "Cherry"]
```

Comment:
This `try!` is on the wrong line, I think?  The array literal doesn't throw, but `keyed(by:)` might.